src/classes/PermanentFileSystem.ts

```diff
@@ -142,6 +143,29 @@ export class PermanentFileSystem {
     return [];
   }
 
+  public async makeDirectory(requestedPath: string): Promise<Folder> {
+    if (isRootPath(requestedPath)) {
+      throw new Error('You cannot create new root level folders via SFTP.');
+    }
+    if (isArchiveCataloguePath(requestedPath)) {
+      throw new Error('You cannot create new archives via SFTP.');
+    }
+    if (isArchivePath(requestedPath)) {
+      throw new Error('You cannot create new folders at the root level of an archive via SFTP.');
+    }
+    const parentPath = path.dirname(requestedPath);
+    const childName = path.basename(requestedPath);
+    const parentFolder = await this.loadFolder(parentPath);
+    this.folderCache.delete(parentPath);
```

Comment:
I guess the reason for this, is to only cache paths that the user has explicitly loaded not, just ones that were loaded due to their own creation?

Reply:
The cache needs to be dirtied because it is no longer accurate -- the cached version of the folder doesn't know about the newly created folder.

I am going to make an issue about this but I fear that the way rclone parallelizes it will result in multiple connections (and therefore multiple caches).

Reply:
Issue created: https://github.com/PermanentOrg/sftp-service/issues/76

Reply:
+1